rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/NodesLockRestorationManager.java

```diff
@@ -0,0 +1,171 @@
+import org.ow2.proactive.resourcemanager.db.RMDBManager;
+import org.ow2.proactive.resourcemanager.rmnode.RMNode;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * Encapsulates the logic about Nodes lock restoration upon Resource Manager restart.
+ * <p>
+ * It was decided to keep the mechanism simple by considering the node source name only
+ * when the lock restoration is made. The hostname does not enter into account.
+ * <p>
+ * Considering the hostname is non-trivial since it is not stable. It may change on
+ * machine restart or due to node state change. Besides, some infrastructures are
+ * dynamics and the number of nodes deployed from a restart to another may differ.
+ * <p>
+ * This class is not thread-safe.
+ *
+ * @author ActiveEon Team
+ * @since 01/02/17
+ */
+public class NodesLockRestorationManager {
+
+    private static final Logger log = Logger.getLogger(NodesLockRestorationManager.class);
+
+    private final RMCore rmCore;
+
+    protected Map<String, MutableInteger> nodeLockedOnPreviousRun;
+
+    protected boolean initialized;
+
+    NodesLockRestorationManager(RMCore rmCore) {
+        this.nodeLockedOnPreviousRun = Maps.newHashMap();
+        this.rmCore = rmCore;
+    }
+
+    protected void initialize() {
+        Stopwatch stopwatch = null;
+
+        if (log.isInfoEnabled()) {
+            stopwatch = Stopwatch.createStarted();
+        }
+
+        nodeLockedOnPreviousRun = findNodesLockedOnPreviousRun();
+
+        if (log.isInfoEnabled()) {
+            stopwatch.stop();
+            log.info("Identifying nodes locked on the previous run required " + stopwatch.elapsed(TimeUnit.MILLISECONDS) +
+                     " ms");
+        }
+
+        if (nodeLockedOnPreviousRun.isEmpty()) {
+            log.info("There is no locks to restore");
+        } else {
+            log.info("Here is the number of nodes to lock per node source:");
+
+            for (Map.Entry<String, MutableInteger> entry : nodeLockedOnPreviousRun.entrySet()) {
+                log.info("  - nodeSource=" + entry.getKey() + ", host=" + entry.getKey() + ", count=" +
+                        entry.getValue().getValue());
+            }
+        }
+
+        initialized = true;
+    }
+
+    Map<String, MutableInteger> findNodesLockedOnPreviousRun() {
+        RMDBManager dbManager = rmCore.getDbManager();
+        Map<String, MutableInteger> nodesLockedOnPreviousRun = dbManager.findNodesLockedOnPreviousRun();
+        dbManager.clearLockHistory();
+        return nodesLockedOnPreviousRun;
+    }
+
+    Map<String, MutableInteger> getNodeLockedOnPreviousRun() {
+        return nodeLockedOnPreviousRun;
+    }
+
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    /**
+     * Handle the specified node for lock restoration.
+     *
+     * @param node the node to consider.
+     */
+    public void handle(RMNode node) {
+
+        if (!initialized || node.isLocked() || isRestorationCompleted()) {
```

Comment:
Could you wrap this validation into a private method that could be called `boolean isNodeValidToBeRestored(RMNode node)` or something like that?